pkg/auth/providers/goauth/goauth_client.go

```diff
@@ -0,0 +1,81 @@
+package goauth
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	googleAPI        = "https://www.googleapis.com/admin/directory/v1/"
+	testgooglegroups = "https://www.googleapis.com/auth/"
+)
```

Comment:
Instead oh hard-coding the endpoints here, maybe its better to add these as defaults on google authconfig crd, and drop from schema so it is not available for API. This way if these change, we can change them on the crd's without needing a code change. Other option would be adding them as setting. I guess this also applies to github auth.

Reply:
@prachidamle I have 3 commits, one separate for the changes done to storing auth config fields as secrets, and because of that these 2 endpoints are showing up although they aren't in the final pr. (I'll squash them now). There's only one hardcoded endpoint and that's the userinfo one in the comment below. I agree that if that endpoint changes we'll have to do code change. But I'm not sure I follow how adding this field on the api will help, esp if we drop it from schema, editing via kubectl will be the only option then?

Reply:
yes natively via kubectl, we dont expect these will change often.

Reply:
Can any user access the authconfig on /v3 API? If it is just global admin then maybe we can add that to API

Reply:
Discussed offline and decided to go ahead with a field on the CRD and not dropping it from the schema, similar to Azure AD's field